### PR TITLE
Tforkner/4196 sharepoint endpoints

### LINKED ISSUE #4196: [Feature] Create endpoints to be used by SharePoint
### Description and related issues
We need two new endpoints for SharePoint to use for automated submission. 
1) endpoint to retrieve APDs that have status submitted but not yet received (double check status language)
2) endpoint for SharePoint to push workflow updates to specific APDs.

### This task is done when…

- [ ] getSubmissions endpoint is created
- [ ] updateReviewStatus endpoing is created
- [ ] swagger is updated with documentation
- [ ] tests are created
- [ ] an access role has been created for SharePoint

### PR DESCRIPTION
**Pull request by thetif**
Resolves #4196

### Description
Creates two endpoints to be used by Kong

### Automated test cases written
Created standard endpoint and unit test.

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| submitted APDs | GET all submitted APDs | returns all APDs with submitted status | tap |
| status updates for APDs | the updates are empty | it handles that gracefully | tap |
| status updates for APDs | there is one valid update | the APD status is updated | tap |
| status updates for APDs | there are multiple updates | the APD statuses are updated | tap |
| status updates for APDs | there are valid and invalid statuses | it handles it gracefully, the valid status is updated and the invalid isn't | tap |
| a GET call to /apds/submissions | the user IP doesn't match | a 403 is returned | jest |
| a GET call to /apds/submissions | the user IP does match | the APDs and a status 200 is returned | jest |
| a PATCH call to /apds/submissions | the user IP doesn't match | a 403 is returned | tap |
| a PATCH call to /apds/submissions | the user IP does match | a 403 is returned | tap |
| a PATCH call to /apds/submissions | the user IP does match but the body is invalid | a 400 is returned | tap |
| a PATCH call to /apds/submissions | the user IP does match and the body is valid | a 204 is returned | tap |
| a PATCH call to /apds/submissions | the user IP doesn't match | a 403 is returned | jest |
| a PATCH call to /apds/submissions | the user IP does match but the body is invalid | a 400 is returned | jest |
| a PATCH call to /apds/submissions | the user IP does match and the body is valid | a 304 is returned | jest |

### Steps to manually verify this change

1. using Postman, GET /apds/submissions and verify that you get a 403
2. using Postman, PATCH /apds/submissions and verify that you get a 403
3. add you public IP to the flag in LaunchDarkly
4. using Postman, GET /apds/submissions and verify that you get a 200
5. using Postman, PATCH /apds/submissions with no body and verify that you get a 400
6. using Postman, PATCH /apds/submissions with a raw json body and verify that you get a 204, the valid apdId should be replaced by an actual apdId which you can find by opening any apd in the browser and copying the long alphanumberic string after apd/ in the url
```    
    [{
        "apdId": "valid apdId",
        "newStatus": "approved"
    }]
```

### This pull request is ready to code review when

- [x] Automated tests are updated (and all tests are passing)
- [x] New automated test cases are documented above
- [x] Pull request has been labeled, if applicable with feature, content, bug,
      tests, refactor
- [x] Associated OpenAPI documentation has been updated

### This pull request is ready to test when

- [x] Code has been reviewed by someone other than the original author

### This pull request is ready to review when the QA has

- [ ] Verified the functionality related to the change
- [ ] Verified that the change works with Narrator on Windows
- [ ] Verified that the change works with VoiceOver on Mac
- [ ] Verified all updated pages with the WAVE tool
- [ ] Verified tab and keyboard navigation functionality

### This pull request can be merged when

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: api/routes/apds/submissions/openApi.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  schema: { arrayOf, jsonResponse }
+} = require('../../openAPI/helpers');
+
+const openAPI = {
+  '/apds/submissions': {
+    get: {
+      tags: ['APDs'],
+      summary: 'Get all of the APDs that have been submitted',
+      description: 'Get a list of all APDs that have been submitted',
+      // security: {
```
Comment (tbolt) [inline api/routes/apds/submissions/openApi.js:11]: What's the plan with the?

**Review by tbolt — APPROVED** [review]

File: api/routes/apds/submissions/openApi.js (hunk 1 above)
Comment (thetif) [reply api/routes/apds/submissions/openApi.js:11]: eventually we will have an api key secure this endpoint but that relies on getting everything working with kong

**Review by tbolt — APPROVED** [review]

**Review by mirano-darren — APPROVED** [review]
Looks good to me! Postman is showing the expected behavior. 
Note: Used PATCH instead of POST

File: api/unit-test.sh [hunk 2]
```diff
@@ -1,27 +1,11 @@
...
-echo 'Checking to see if the server is running'
-until [ "`docker inspect -f {{.State.Health.Status}} api-container`"=="healthy" ]; do
-    sleep 10;
-    echo '.';
-done;
-echo 'Server is running and status is healthy'
 docker-compose -f ../docker-compose.endpoint-tests.yml -p api exec api-for-testing yarn run test $@
-echo $?
-EXIT_CODE=$?
+EXIT_CODE=$? # this must stay right after the test run so that the EXIT_CODE is properly reported
```
Comment (tbolt) [inline api/unit-test.sh:6]: 😃

File: api/unit-test.sh [hunk 3]
```diff
@@ -1,27 +1,11 @@
 #!/usr/bin/env sh
 
-export NODE_ENV=test
+./test-server-setup.sh
```
Comment (tbolt) [inline api/unit-test.sh:3]: awesome!

File: api/db/apds.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 const { applyPatch } = require('fast-json-patch');
+// const ObjectId = require('mongoose').Types.ObjectId;
```
Comment (tbolt) [inline api/db/apds.js:2]: did you want to keep this in?

File: api/routes/apds/post.test.js [hunk 5]
```diff
@@ -296,146 +296,181 @@ tap.test('apds POST endpoint', async endpointTest => {
...
-            apdOverview: {
-              updateStatus: {
-                isUpdateAPD: true,
-                annualUpdate: false,
-                asNeededUpdate: true
-              }
-            }
-          }
-        };
+    // tests.test(
```
Comment (tbolt) [inline api/routes/apds/post.test.js:299]: we going to revisit this later?

**Review by tbolt — APPROVED** [review]
Resolves many things 🎉

**Review by tbolt — APPROVED** [review]

**Review by tbolt — APPROVED** [review]

**Review by mirano-darren — APPROVED** [review]
LGTM!

File: api/routes/apds/post.data.js [hunk 6]
```diff
@@ -1,23 +1,22 @@
...
-const getNewApd = apdType => {
-  const years = defaultAPDYears();
+const getNewApd = (apdType, years = [...defaultAPDYears()]) => {
   const yearOptions = defaultAPDYearOptions();
 
   let specific = {};
   switch (apdType) {
     case APD_TYPE.HITECH: {
-      specific = getNewHitechApd();
+      specific = getNewHitechApd(years, yearOptions);
```
Comment (tbolt) [inline api/routes/apds/post.data.js:15]: 👍

**Review by tbolt — APPROVED** [review]
Looks good

**Review by jeromeleecms — APPROVED** [review]
Once the build passes, I'm good with this. I don't have a way to test the endpoints easily - is there an automated way to report on the acceptance criteria that was established? Or is that embedded into the testing?

**Review by tbolt — APPROVED** [review]

**Pull request merged**